code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -615,12 +615,15 @@
 	if(iscarbon(M) && M.stat != DEAD)
 		var/mob/living/carbon/C = M
 		var/holycheck = ishumanbasic(C)
-		if(!(holycheck || islizard(C)) || reac_volume < 5) // implying xenohumans are holy //as with all things,
+		if(!(holycheck || islizard(C) || ismoth(C)) || C.dna.features["moth_wings"] == "Burnt Off" || reac_volume < 5 ) // implying xenohumans are holy //as with all things,
 			if(method == INGEST && show_message)
 				to_chat(C, "<span class='notice'><i>You feel nothing but a terrible aftertaste.</i></span>")
 			return ..()
-
-		to_chat(C, "<span class='userdanger'>A terrible pain travels down your back as wings burst out!</span>")
+		if(ismoth(C))
+			to_chat(C, "<span class='userdanger'>A terrible pain travels down your back as your wings change shape!</span>")
+			C.dna.features["moth_wings"] = "None"
+		else
+			to_chat(C, "<span class='userdanger'>A terrible pain travels down your back as wings burst out!</span>")
```

Comment:
A generic `HAS_WINGS` flag or variable would be nice here.

Reply:
By this do you mean you'd like it to check if they have a moth_wings, assigning that to a variable, and then checking that, instead of checking if they're a moth?

Reply:
I mean to perform generic behavior. If the species has innate wings of any kind then the feedback is that the wings change. Makes sense?

Reply:
In the context of making a proc to check if the user is a valid target for a flight potion I think so. 
I don't think I ever considered that we might end up with something beyond mothpeople that have innate wings.